### PR TITLE
fix: APP-3070 - Update WC network switch flow via dAppConnect

### PR DESCRIPTION
**Pull request by thekidnamedkd**
## Description

proposed solution to align the WalletConnect connector with the current app network (chainId) across the app

Task: [APP-3070](https://aragonassociation.atlassian.net/browse/APP-3070)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I have tested my code on the test network.

[APP-3070]: https://aragonassociation.atlassian.net/browse/APP-3070?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/hooks/useWallet.tsx [hunk 1]
```diff
@@ -71,6 +71,18 @@ export const useWallet = (): IUseWallet => {
     [network, signer?.provider]
   );
 
+  // console.log left for PR conversation purposes
+  console.log(
```
Comment (sepehr2github) [inline src/hooks/useWallet.tsx]: Do not forget to remove this. 😉

File: src/app.tsx [hunk 2]
```diff
@@ -38,12 +38,29 @@ import {GatingMenu} from 'containers/gatingMenu';
...
+  const {methods, status, network, address, provider, connectorName} =
+    useWallet();
+  const connectors = useConnectors();
+
+  // TODO explore more sturdy ways to handle this in the actual payload and/or a UI flow for network "switching"
+  // Update incoming dAppConnect-as-wallet's current chain to the chainId of the app's current req'd network
+  // All chains when connecting to the app are pre approved by WalletConnect on connection
+  // This is more of a workaround to match chains and unblock existing flows
+  // Ex. Aragon DAO connecting to an Aragon DAO, the connecting DAO is on the wrong chain (default chainId of WAGMI config -- was Base)
+  useEffect(() => {
```
Comment (cgero-eth) [inline src/app.tsx]: What if, instead, we update the `NetworkErrorMenu` dialog [here](https://github.com/aragon/app/blob/develop/src/containers/networkErrorMenu/index.tsx#L89) to display the switch network for whatever connector being used?
Users will need to click on the menu and then on switch network, but we would avoid this global use-effect

**Comment by thekidnamedkd** [comment]
@sepehr2github @cgero-eth it's kind of a long set up for this scenario, but as evidence here's the console before and after clicking 'switch network' in the network error modal:

image

File: src/hooks/useSwitchNetwork.tsx [hunk 3]
```diff
@@ -44,6 +48,20 @@ export const useSwitchNetwork = () => {
         'MetaMask is not installed. Please consider installing it: https://metamask.io/download.html'
       );
     }
+    // TODO explore more sturdy ways to handle this in the actual payload and/or a UI flow for network "switching"
+    // Update incoming dAppConnect-as-wallet's current chain to the chainId of the app's current req'd network
+    // All chains when connecting to the app are pre approved by WalletConnect on connection
+    // This is more of a workaround to match chains and unblock existing flows
+    // Ex. Aragon DAO connecting to an Aragon DAO, the connecting DAO is on the wrong chain (default chainId of WAGMI config -- was Base)
+    if (connectorName === 'WalletConnect') {
```
Comment (cgero-eth) [inline src/hooks/useSwitchNetwork.tsx]: If we let the users switch the network through the network-error-menu this is not needed, right?

File: src/containers/networkErrorMenu/index.tsx [hunk 4]
```diff
@@ -86,7 +86,8 @@ export const NetworkErrorMenu = () => {
             </Trans>
           </WarningDescription>
         </WarningContainer>
-        {connectorName === 'MetaMask' && (
+        {(connectorName === 'MetaMask' ||
+          connectorName === 'WalletConnect') && (
```
Comment (cgero-eth) [inline src/containers/networkErrorMenu/index.tsx]: I believe we can always display the switch-network button

File: src/app.tsx [hunk 5]
```diff
@@ -38,12 +38,19 @@ import {GatingMenu} from 'containers/gatingMenu';
...
   // TODO this needs to be inside a Routes component. Will be moved there with
   // further refactoring of layout (see further below).
   const {pathname} = useLocation();
   const {methods, status, network, address, provider} = useWallet();
+  const connectors = useConnectors();
+
+  console.log(connectors);
+  const wcConnector = connectors.filter(c => c.name === 'WalletConnect');
+
+  console.log('wcConnector', wcConnector);
```
Comment (cgero-eth) [inline src/app.tsx]: We can remove these console logs

**Review by cgero-eth — APPROVED** [review]
Much cleaner now by using wagmi functionalities, thanks @thekidnamedkd! Let's only remove the console logs and we can merge the PR

File: src/app.tsx (hunk 5 above)
Comment (thekidnamedkd) [reply src/app.tsx]: No use for the connectors hook, wcConnector filter either. Removed as well.

**Pull request merged**